sklearn/compose/tests/test_column_transformer.py

```diff
@@ -249,6 +249,14 @@ def transform(self, X, y=None):
     assert ct.transformers_[-1][1] == 'drop'
     assert_array_equal(ct.transformers_[-1][2], [1])
 
+    assert_array_equal(ct.fit_transform(X_df2), X_res_first)
```

Comment:
This is duplicated from above

Reply:
The above lines created df2 with integer column names, but never used it in the test (which I assumed was a typo). So I have duplicated the above tests using df2 instead of df instead of replacing df with df2 just incase it wasn't a typo.

Reply:
That indeed looks like a typo, but I think you can rather correct the code above, than duplicating it (since it is not related to the rest of this PR, also feel free to open a separate PR for this)

Reply:
Okay, I will remove the duplication and submit this as a separate pull request, thanks

Reply:
Raised #14128 for this instead